src/client/Microsoft.Identity.Client/ManagedIdentity/ManagedIdentityClient.cs

```diff
@@ -60,29 +60,35 @@ internal static ManagedIdentitySource GetManagedIdentitySource(ILoggerAdapter lo
             string msiEndpoint = EnvironmentVariables.MsiEndpoint;
             string msiSecretMachineLearning = EnvironmentVariables.MsiSecret;
             string imdsEndpoint = EnvironmentVariables.ImdsEndpoint;
-            string podIdentityEndpoint = EnvironmentVariables.PodIdentityEndpoint;
+
+            logger?.Info("[Managed Identity] Detecting managed identity source...");
```

Comment:
Why can't we guarantee that the logger is not null?

Reply:
GetManagedIdentitySource() is exposed as a public API on MIA, and the logger can be null. We cannot guarantee that logger will always be non-null here